--- Prelude_Bool.thy.orig	2008-05-05 14:18:17.000000000 -0300
+++ Prelude_Bool.thy	2008-05-05 14:18:17.000000000 -0300
@@ -44,12 +44,18 @@
 declare And_def4 [simp]
 
 theorem Not_True_1 : "ALL x. Not' x = True' = (x = False')"
-using Or_def Otherwise_def by auto
+apply auto
+apply(case_tac x)
+apply auto
+done
 
 ML "Header.record \"Not_True_1\""
 
 theorem Not_False_2 : "ALL x. Not' x = False' = (x = True')"
-using Or_def Otherwise_def by auto
+apply auto
+apply(case_tac x)
+apply auto
+done
 
 ML "Header.record \"Not_False_2\""
 
