--- Prelude_Bool.thy.orig	2008-08-21 15:34:04.000000000 -0300
+++ Prelude_Bool.thy	2008-08-21 15:34:04.000000000 -0300
@@ -40,22 +40,34 @@
 declare AndTrue [simp]
 
 theorem NotFalse1 : "ALL x. Not' x = True' = (x = False')"
-by auto
+apply auto
+apply(case_tac x)
+apply auto
+done
 
 ML "Header.record \"NotFalse1\""
 
 theorem NotTrue1 : "ALL x. Not' x = False' = (x = True')"
-by auto
+apply auto
+apply(case_tac x)
+apply auto
+done
 
 ML "Header.record \"NotTrue1\""
 
 theorem notNot1 : "ALL x. (~ x = True') = (Not' x = True')"
-by auto
+apply(auto)
+apply(case_tac x)
+apply(auto)
+done
 
 ML "Header.record \"notNot1\""
 
 theorem notNot2 : "ALL x. (~ x = False') = (Not' x = False')"
-by auto
+apply(auto)
+apply(case_tac x)
+apply(auto)
+done
 
 ML "Header.record \"notNot2\""
 
